vscode/DscExt/src/extension.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+
+export function activate(context: vscode.ExtensionContext) {
+
+	const dsc_type_provider = vscode.languages.registerCompletionItemProvider({ language: 'yaml', pattern: '**/*.dsc.yaml' },
+		{
+			provideCompletionItems(document: vscode.TextDocument, position: vscode.Position) {
+
+				const linePrefix = document.lineAt(position).text.slice(0, position.character);
+				if (!linePrefix.endsWith('type:')) {
+					return undefined;
+				}
+
+				let ps_cache_file_path: string = '';
+				if (process.platform==='win32')
+				{
+					ps_cache_file_path = process.env.LocalAppData + '\\dsc\\PSAdapterCache.json';
+				}
+				else
+				{
+					ps_cache_file_path = process.env.HOME + '/.dsc/PSAdapterCache.json';
+				}
+				console.log("Using cache path " + ps_cache_file_path);
+
+				if (!fs.existsSync(ps_cache_file_path)) {
+					console.log("Cache file does not exist");
+					return [];
+				}
+
+				const dataArray = JSON.parse(fs.readFileSync(ps_cache_file_path, 'utf-8'));
```

Comment:
What's the reason we made this synchronous? If not let's use `readFile` from Node's `fs/promises` and get this code back to being async and non-blocking (ditto above with the exist check).